extensions/apps/antenna/src/extensions/beam-editor/beam-editor.tsx

```diff
@@ -264,8 +296,40 @@ export const BeamEditor: React.FC = () => {
     });
   }, [blocksInUse, focusedBlock]);
 
+  const handleLeavePage = () => {
+    /**
+     * get the appName and navigationUrl from the newUrl state
+     */
+    const splitPathNames = new URL(newUrl).pathname.split('/');
+    const appName = splitPathNames.slice(1, 3).join('/');
```

Comment:
appName here is not correct, you are assuming that names are always in the form of `<namespace>/<appname>` which is not the case.
For this specific case you'll need to use singleSpa.navigateToUrl. you have access to `singleSpa` from the `useRootComponentProps`